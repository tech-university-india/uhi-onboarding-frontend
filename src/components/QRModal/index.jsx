import React from "react";
const QRModal = ({ visible, onClose }) => {
  const handleClose = (e) => {
    if (e.target.id === "container") onClose();
  };
  if (!visible) return null;
  const qrCode = '64base-imageformat'
  return (
    <div
      id="container"
      onClick={handleClose}
      data-testid="container"
      className="fixed inset-0 bg-black bg-opacity-10 backdrop-blur-sm flex justify-center items-center p-4"
    >
      <div className="bg-white w-96 h-96 rounded-lg shadow-lg flex flex-col justify-center items-center">
        <div className="w-80 h-80 bg-primary rounded-lg flex justify-center items-center">
          <img src={`data:image/jpeg;base64,${qrCode}`} alt="qr-code" className="w-64 h-64" />
        </div>
      </div>
    </div>
  );
};

export default QRModal;
