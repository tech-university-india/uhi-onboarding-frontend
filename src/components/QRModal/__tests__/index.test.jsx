import QRModal from ".."
import { render,screen,fireEvent } from "@testing-library/react"
describe('QR Modal', () => {
    it('should render component', () => {
        const {asFragment} = render(<QRModal />)
        expect(asFragment()).toMatchSnapshot()
    })
    it('should render component with props', () => {
        const onClose = jest.fn()
        render(<QRModal visible onClose={onClose} />)
        expect(screen.getByTestId('container')).toBeInTheDocument()
    })
    it('should call onClose when click on container', () => {
        const onClose = jest.fn()
        render(<QRModal visible onClose={onClose} />)
        fireEvent.click(screen.getByTestId('container'))
        expect(onClose).toHaveBeenCalled()
    })

})