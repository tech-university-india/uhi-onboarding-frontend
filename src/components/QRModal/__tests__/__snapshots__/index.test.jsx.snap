// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QR Modal should render component 1`] = `<DocumentFragment />`;
